vectorstores/chromadb/chroma.go

```diff
@@ -0,0 +1,187 @@
+package chromadb
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	chroma "github.com/amikos-tech/chroma-go"
+	chromaopenapi "github.com/amikos-tech/chroma-go/swagger"
+	"github.com/tmc/langchaingo/schema"
+	"github.com/tmc/langchaingo/vectorstores"
+)
+
+var (
+	// ErrEmbedderWrongNumberVectors is returned when if the embedder returns a number
+	// of vectors that is not equal to the number of documents given.
+	ErrEmbedderWrongNumberVectors = errors.New(
+		"number of vectors from embedder does not match number of documents",
+	)
+	ErrInvalidScoreThreshold = errors.New(
+		"score threshold must be between 0 and 1")
+)
+
+// Store is a wrapper around the chromadb client.
+type Store struct {
+	embedder wrappedEmbeddingFunction
+	client   *chroma.Client
+
+	scheme string
+	host   string
+
+	// optional
+	nameSpaceKey string
+	// optional
+	textKey string
+	// optional: nameSpace represents collection in chromadb
+	nameSpace string
+	// optional
+	distanceFunc chroma.DistanceFunction
+	// optional
+	includes []chroma.QueryEnum
+}
+
+var _ vectorstores.VectorStore = Store{}
+
+// New creates a new Store with options for chromadb.
+func New(opts ...Option) (vectorstores.VectorStore, error) {
+	s, err := applyClientOptions(opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	configuration := chromaopenapi.NewConfiguration()
+	configuration.Servers = chromaopenapi.ServerConfigurations{
+		{
+			URL:         fmt.Sprintf("%s://%s", s.scheme, s.host),
+			Description: "Chromadb server url for this store",
+		},
+	}
+	s.client = &chroma.Client{
+		ApiClient: chromaopenapi.NewAPIClient(configuration),
+	}
+
+	if _, err = s.client.Heartbeat(); err != nil {
```

Comment:
At first, I suspected your calling the `Heartbeat` method here was new behavior, as I thought the other `New` functions (i.e., of the other `vectorstore` packages) didn't actually do any network I/O.  Reviewing them though I realized they _can_ initiate some (even if only asynchronously).

It might be a good idea to reflect in the comments/doc that this function actually makes a network call and will fail if immediately it can't reach the external service to make that clearer.